RecoEgamma/EgammaElectronAlgos/src/PixelHitMatcher.cc

```diff
@@ -106,52 +114,79 @@ PixelHitMatcher::compatibleSeeds
   
   std::vector<SeedWithInfo> result ;
   
-  mapTsos_fast_.clear();  
+  //mapTsos_fast_.clear();  
   mapTsos2_fast_.clear();  
-  mapTsos_fast_.reserve(seeds->size()) ;
+  // mapTsos_fast_.reserve(seeds->size()) ;
   mapTsos2_fast_.reserve(seeds->size()) ;
 
+  // std::vector<TrajectoryStateOnSurface> vTsos(theTrackerGeometry->dets().size());
+  // TrajectoryStateOnSurface vTsos[theTrackerGeometry->dets().size()];
+
+  auto ndets = theTrackerGeometry->dets().size();
+  // std::vector<bool> away(ndets,false);
+
+  int iTsos[ndets];
+  for ( auto & i : iTsos) i=-1;
+  std::vector<TrajectoryStateOnSurface> vTsos; vTsos.reserve(seeds->size());
+
   for(const auto& seed : *seeds) {
     hit_gp_map_.clear();
     if( seed.nHits() > 9 ) {
       edm::LogWarning("GsfElectronAlgo|UnexpectedSeed") <<"We cannot deal with seeds having more than 9 hits." ;
       continue;
     }
+    
     const TrajectorySeed::range& hits = seed.recHits();
     // cache the global points
+   
     for( auto it = hits.first; it != hits.second; ++it ) {
       hit_gp_map_.emplace_back(it->globalPosition());      
     }
-    //iterate on the hits    
-    for( auto it1 = hits.first; it1 != hits.second; ++it1 ) {
+
+    //iterate on the hits 
+    auto he =  hits.second -1;   
+    for( auto it1 = hits.first; it1 < he; ++it1 ) {
       if( !it1->isValid() ) continue;
-      const unsigned idx1 = std::distance(hits.first,it1);
+      auto  idx1 = std::distance(hits.first,it1);
       const DetId id1 = it1->geographicalId();
-      const GeomDet *geomdet1 = it1->det();      
+      const GeomDet *geomdet1 = it1->det();
+
+      auto ix1 = geomdet1->gdetIndex();
+      assert(ix1<int(ndets));
+      /*
+      if (away[ix1]) continue;
+      away[ix1] = geomdet1->position().basicVector().dot(xmeas.basicVector()) <0;
+      if (away[ix1]) continue;
+      */
+
       const GlobalPoint& hit1Pos = hit_gp_map_[idx1];
+      auto dt = hit1Pos.x()*xmeas.x()+hit1Pos.y()*xmeas.y();
+      if (dt<0) continue;
+      if (dt<phicut*(xmeas_r*hit1Pos.perp())) continue;
 
-      const TrajectoryStateOnSurface* tsos1;      
-      DetTsosAssoc::iterator tsos1_itr = mapTsos_fast_.find(geomdet1);
-      if( tsos1_itr != mapTsos_fast_.end() ) {	
-	tsos1 = &(tsos1_itr->second);
-      } else {
-	auto empl_result =
-	  mapTsos_fast_.emplace(geomdet1,prop1stLayer->propagate(tsos,geomdet1->surface()));
-	tsos1 = &(empl_result.first->second);
+      if(iTsos[ix1]<0)   {
+        iTsos[ix1] = vTsos.size();
+        vTsos.push_back(prop1stLayer->propagate(tsos,geomdet1->surface()));
       }
+      auto tsos1 = &vTsos[iTsos[ix1]];
+
       if( !tsos1->isValid() ) continue;
       std::pair<bool, double> est = ( id1.subdetId() % 2 ? 
 				      meas1stBLayer.estimate(vprim, *tsos1, hit1Pos) :
 				      meas1stFLayer.estimate(vprim, *tsos1, hit1Pos)  );
       if( !est.first ) continue;
-      if( std::abs(normalized_phi(hit1Pos.phi()-xmeas_phi))>2.5 ) continue;
```

Comment:
ah, this is the 2.5 from https://github.com/cms-sw/cmssw/pull/13819/files#r57508991
.. OK then.